lib/ansible/modules/cloud/amazon/ec2_instance_facts.py

```diff
@@ -500,8 +500,11 @@ def list_ec2_instances(connection, module):
 
     # Turn the boto3 result in to ansible friendly tag dictionary
     for instance in snaked_instances:
-        if 'tags' in instance:
-            instance['tags'] = boto3_tag_list_to_ansible_dict(instance['tags'])
+        if 'tags' in instance and len(instance['tags']) > 0:
```

Comment:
Seems simpler to just do:

```
instance['tags'] = boto3_tag_list_to_ansible_dict(instance.get('tags', []), 'key', 'value')
```

This way, we always return `tags`. There doesn't seem to be a need to check whether `key` is in `instance['tags'][0]` as we've just snaked it.

Reply:
Thank you for the feedback. 
You are right. Forgot about the snaking.. Made the change and pushed